dimod/quadratic/quadratic_model.py

```diff
@@ -1224,5 +1224,37 @@ def Integers(labels: Union[int, Iterable[Variable]],
     else:
         yield from (Integer(dtype=dtype) for _ in range(labels))
 
+
+def IntegerArray(labels: Union[int, Iterable[Variable]],
+                 dtype: Optional[DTypeLike] = None) -> np.ndarray:
+    """Return a numpy array of quadratic models, each with a 
+    single integer variable.
+
+    Args:
+        labels: Either an iterable of variable labels or a number. If a number
+            labels are generated using :class:`uuid.UUID`.
+        dtype: Data type for the returned quadratic models.
+
+    Returns:
+        Array of quadratic models, each with a single integer variable.
+    
+    """
+    return _VariableArray(Integers, labels, dtype)
+
+
+def _VariableArray(variable_generator: Callable,
+                   labels: Union[int, Iterable[Variable]],
+                   dtype: Optional[DTypeLike] = None) -> np.ndarray:
+    """Builds numpy array from a variable generator method."""
+    if isinstance(labels, int):
+        number_of_elements = labels
+    else:
+        number_of_elements = len(labels)
```

Comment:
This will currently fail for `labels = (v for v in range(5))` which is an iterable, but is not sized. One thing you could do is
```python
if isinstance(labels, int):
    number_of_elements = labels
elif isinstance(labels, Sized):
    number_of_elements = len(labels)
else:
    labels = list(labels)
    number_of_elements = len(labels) 
```
Dumps it to memory which is not ideal but I can't think of a nice way around it.

Reply:
At first glance the edge case looked like tuple comprehension, but now I see that's actually done with `tuple(...)`, and would be sized. Thanks for pointing this out!